apps/solaredgemonitor/solaredge_monitor.star

```diff
@@ -5,167 +5,671 @@ Description: Energy production and consumption monitor for your SolarEdge solar
+""")
+
+def render_fail(rep):
+    content = json.decode(rep.body())
+    return render.Root(render.Box(render.WrappedText(str(rep.status_code) + " : " + content["String"], color = RED)))
+
+def get_energy_details(site_id, api_key, tz, interval):
+    now = time.now().in_location(tz)
+    start_string = ""
+    now_string = humanize.time_format("yyyy-MM-dd HH:mm:ss", now)
+    if interval == "day":  # from the start of today to now, just set time to 00:00
+        start_string = humanize.time_format("yyyy-MM-dd 00:00:00", now)
+        now_string = humanize.time_format("2006-01-02 15:04:05", now)
+        time_unit = "DAY"
+    elif interval == "24h":  # the last 24hr from now, use a now-duration
+        duration = time.parse_duration("24h")
```

Comment:
should be some way to get a time constant without parsing

Reply:
Done.